### ISSUE TITLE
index of gif may wrong after downloaded

### ISSUE THREAD

**Issue by wuyongrui** [rs/SDWebImage#1497]
```
- (void)sd_setAnimationImagesWithURLs:(NSArray *)arrayOfURLs {
    [self sd_cancelCurrentAnimationImagesLoad];
    __weak __typeof(self)wself = self;

    NSMutableArray *operationsArray = [[NSMutableArray alloc] init];

    for (NSURL *logoImageURL in arrayOfURLs) {
        NSLog(@"--start download-> %@",logoImageURL);
        id <SDWebImageOperation> operation = [SDWebImageManager.sharedManager downloadImageWithURL:logoImageURL options:0 progress:nil completed:^(UIImage *image, NSError *error, SDImageCacheType cacheType, BOOL finished, NSURL *imageURL) {
            if (!wself) return;
            NSLog(@"--finish download-> %@",imageURL);
            dispatch_main_sync_safe(^{
                __strong UIImageView *sself = wself;
                [sself stopAnimating];
                if (sself && image) {
                    NSMutableArray *currentImages = [[sself animationImages] mutableCopy];
                    if (!currentImages) {
                        currentImages = [[NSMutableArray alloc] init];
                    }
                    [currentImages addObject:image];

                    sself.animationImages = currentImages;
                    [sself setNeedsLayout];
                }
                [sself startAnimating];
            });
        }];
        [operationsArray addObject:operation];
    }

    [self sd_setImageLoadOperation:[NSArray arrayWithArray:operationsArray] forKey:@"UIImageViewAnimationImages"];
}
```

sorry for the pool english。

just like i set the urls as [a,b,c,d], and the d is smaller,so the finish download may like [d,a,b,c].
another issue:  maybe c is download fail,it will blink, 
maybe it should startAnimating while all download success and with the origin index of url.

**Comment by mythodeia** [comment]
so you say we put the ```[sself startAnimating];``` inside the ``` if (sself && image) {``` 
to make sure that we have an image to animate before we do it, right?

**Comment by wuyongrui** [comment]
I think it can use A counter or something to make sure it  have download
all the images and then starting animation
2016年3月15日 下午10:54，"Konstantinos K." <notifications@github.com>写道：

**Comment by bpoplauschi** [comment]
Probably related to #1452

**Comment by bpoplauschi** [comment]
Fixed by #1452
